bootloader/src/pyi_archive.c

```diff
@@ -553,7 +553,7 @@ pyi_arch_status_free_memory(ARCHIVE_STATUS *archive_status)
  * for freeing it.
  */
 char *
-pyi_arch_get_option(ARCHIVE_STATUS * status, char * optname)
+pyi_arch_get_option(const ARCHIVE_STATUS * status, char * optname)
```

Comment:
These changes (adding "const") are unrelated to this pull-request and should go into a commit of its own.

Reply:
In fact, these were required, because I added an argument of type `const ARCHIVE_STATUS *` to `pyi_utils_create_child` below.

I can either make these existing methods const-correct or I can make the new argument const-incorrect. I chose the option that I felt was a net improvement. Please let me know if you disagree.